app/src/main/java/org/oppia/app/player/state/hintsandsolution/HintsAndSolutionAdapter.kt

```diff
@@ -0,0 +1,231 @@
+package org.oppia.app.player.state.hintsandsolution
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.RecyclerView
+import org.oppia.app.databinding.HintsSummaryBinding
+import org.oppia.app.databinding.SolutionSummaryBinding
+import org.oppia.util.parser.HtmlParser
+
+// TODO(#216): Make use of generic data-binding-enabled RecyclerView adapter.
+
+private const val TAG_REVEAL_SOLUTION_DIALOG = "REVEAL_SOLUTION_DIALOG"
+private const val VIEW_TYPE_HINT_ITEM = 1
+private const val VIEW_TYPE_SOLUTION_ITEM = 2
+
+/** Adapter to bind StorySummary to [RecyclerView] inside [HintsAndSolutionFragment]. */
+class HintsAndSolutionAdapter(
```

Comment:
Press `Control + Alt + L` - reformatting shortcut to fix the formatting.

Reply:
done.